pandas/core/indexes/base.py

```diff
@@ -4799,11 +4803,23 @@ def _get_engine_target(self) -> ArrayLike:
             return self._values.astype(object)
         return vals
 
+    def _get_join_target(self) -> ArrayLike:
+        """
+        Get the ndarray or ExtensionArray that we can pass to the join
+        functions.
+        """
+        if isinstance(self._values, BaseMaskedArray):
+            # This is only used if our array is monotonic, so no NAs present
```

Comment:
Where is monoticity checked of `self._values._data`?

Reply:
Using those functions is guarded by those checks. The actual join implementations here only work with monotonic data. As soon as we reach ``_inner_indexer`` and others we have to be monotonic

Reply:
Ah okay cool